### PR TITLE
Tim/add articles

### PR DESCRIPTION
**Pull request by tscanlin**
I just finished refactoring the developer docs site to add support for articles. This was a request I got from @pamoptly because we want to include more technical articles on the dev docs instead of in Optiverse. Let me know if anyone else would find this useful. This functionality could also be used to add a blog for instance, but right now I'm just going to be helping Pam set articles up for ios & android.

Even though it seemed like a pretty simple request, the technical challenge was actually kind of tricky. To make this a good experience I refactored the way we compile our html / markdown files to make it far more flexible. I won't get into the weeds, but the change I made could also potentially allow for a full site search to be implemented (see: https://github.com/optimizely/developers.optimizely.com/issues/156).

I would love a technical review before merging these changes. I will do more manual testing myself as well. @bhamodi or @aliabbasrizvi could one of you give this a once over?

CC: @pamoptly @jprovine @Bigspencey @atong518

### CODE REVIEW

File: gulp/tasks/build.js [hunk 1]
```diff
@@ -1,10 +1,9 @@
 var gulp = require('gulp');
 
 gulp.task('build', [
-  'concat',
-  'html-templates',
   'copy-files',
   'sass',
   'browserify',
   'images',
+  'browser-sync-reload', // does 'markdown' and 'html-templates' too.
```
Comment (bhamodi) [inline gulp/tasks/build.js]: Will depending on live reload here cause issues for `npm run deploy`? Currently the deploy script wipes out the `dist` directory and runs `build` without necessarily needing the developer to be running the website locally.

File: src/content/ios/articles/1-article-test.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+includeSiblingData: true
+title: "Article 1 Header"
+summary: |
+  Here is a summary
+  That can go on multiple lines
+---
+
+# Article 1 Header
+
+Some sample text.
```
Comment (bhamodi) [inline src/content/ios/articles/1-article-test.md]: Will these test articles appear anywhere on the website after deploying?

Comment (tscanlin) [inline src/content/ios/articles/1-article-test.md]: Yes, but I have a meeting with @pamoptly tomorrow to add real content for these first few articles. So the content in this PR will be updated before I merge this.

**Comment by bhamodi** [comment]
@tscanlin - Just completed my first pass through this, looks good! :+1: 
I may have some time tomorrow to pull this branch locally and test it out. But in terms of code, it seems good to me.

**Comment by bhamodi** [comment]
Also the potential for search functionality is exciting! :)

**Comment by tscanlin** [comment]
Awesome, thanks @bhamodi!

File: gulp/tasks/build.js (hunk 1 above)
Comment (tscanlin) [inline gulp/tasks/build.js]: That's a good point, I should probably separate this out even though it probably shouldn't hurt, good call.

**Comment by tscanlin** [comment]
@pamoptly I added the articles folder for android too so it should work now. Probably need to restart `npm run gulp` though.

Once you've finished loading in the first few pages of content let me know and I'll give it one more pass before merging / deploying.

Let me know if you need help with anything!

**Comment by pamoptly** [comment]
@tscanlin I updated all the articles - looking good from my side!  I think it's ready to merge - let me know when the merge / deploy happens!

**Comment by tscanlin** [comment]
I just made a few final updates, everything is looking good. I'm gonna merge this.

**Pull request merged**